x-pack/plugins/observability_solution/inventory/public/components/inventory_page_template/index.tsx

```diff
@@ -62,6 +65,23 @@ export function InventoryPageTemplate({ children }: { children: React.ReactNode
     [inventoryAPIClient]
   );
 
+  const isLoadingComplete = useIsLoadingComplete({
```

Comment:
In this case, I chose not to include the logic directly within the component, as it would have made the code in `InventoryPageTemplate` feel more complicated. Keeping it separate seemed like a cleaner and more organized and testable solution.

Since the two loading states present in `InventoryPageTemplate` only work with boolean data and return `false` when the initial request hasn’t been initialized yet, I had to add some extra logic to properly handle when everything was fully loaded.

What approach do you suggest? I might have overcomplicated the solution.

Reply:
`useIsLoadingComplete` works but imho makes things a bit more complicated. 
 
What do you think about having each component responsible for sending its own view state?

Reply:
Thanks for the proposal! However, not all three states are independent React components at the moment, which would require some modifications.

Additionally, to ensure the event is reported only once per view, given the current lifecycle of the components and their boolean loading states we would need to make adjustments to all three components. 

Keeping it as it is allows us to simply create this new hook and implement it in one single component without extensive changes.

I’d be happy to have a quick call about this if you think there’s something I’m missing by not seeing your suggestion as a simpler option than the one I did.

Reply:
oh I see.  Let's proceed with your suggestion.